### PR TITLE
add pre-commit and github action for python formatting. includes .mpc files

### PR DESCRIPTION
**Pull request by eriktaubeneck**
adding in python formatting tools, supporting #54

### CODE REVIEW

**Comment by eriktaubeneck** [comment]
@richajaindce looks like you set this up originally for rust. Can you double check me here for the python setup? thanks!

File: pre-commit [hunk 1]
```diff
@@ -47,3 +47,15 @@ error() {
 cargo clippy --tests -- -D warnings --D clippy::pedantic || error "Clippy errors"
 
 cargo test || error "Test failures"
+
+which flake518 || error "flake518 not found. Install with \`python3 -m pip install flake518\` in the desired python environment."
```
Comment (martinthomson) [inline pre-commit:51]: These might do better with a venv instead.

**Review by martinthomson — APPROVED** [review]

File: pre-commit (hunk 1 above)
Comment (eriktaubeneck) [reply pre-commit:51]: Yea, I thought about that. I figured leaving it up to the user to figure out if they want to use a venv, but we can also go that route.

Comment (eriktaubeneck) [reply pre-commit:51]: @martinthomson  any objection to merging like this, and setting that up if/when we need it? (trying to unblock while i'm on vacation)

**Pull request merged**